src/time_steppers.jl

```diff
@@ -45,237 +42,224 @@ time_step!(model; Nt, Δt, kwargs...) = time_step!(model, Nt, Δt; kwargs...)
+    update_solution!(U, C, arch, grid, Gⁿ, P.pNHS, Δt)
 
     # Recompute vertical velocity w from continuity equation to ensure incompressibility
-    fill_halo_regions!(U, bcs.solution[1:3], arch, grid, boundary_condition_args...)
-    @launch device(arch) config=launch_config(grid, 2) compute_w_from_continuity!(grid, U)
+    fill_halo_regions!(U, model.boundary_conditions.solution[1:3], arch, grid, bc_args...)
+    @launch device(arch) config=launch_config(grid, 2) compute_w_from_continuity!(U, grid)
 
     model.clock.time += Δt
     model.clock.iteration += 1
 
     return nothing
 end
 
-function solve_for_pressure!(::CPU, model::Model)
-    ϕ = model.poisson_solver.storage
-
-    solve_poisson_3d!(model.poisson_solver, model.grid)
-    data(model.pressures.pNHS) .= real.(ϕ)
-end
-
-function solve_for_pressure!(::GPU, model::Model)
-    ϕ = model.poisson_solver.storage
-
-    solve_poisson_3d!(model.poisson_solver, model.grid)
-    @launch device(GPU()) config=launch_config(model.grid, 3) idct_permute!(model.grid, model.poisson_solver.bcs, ϕ,
-                                                                            model.pressures.pNHS.data)
-end
-
-""" Store previous source terms before updating them. """
-function store_previous_source_terms!(grid::AbstractGrid, Gⁿ, G⁻)
-    @loop for k in (1:grid.Nz; (blockIdx().z - 1) * blockDim().z + threadIdx().z)
-        @loop for j in (1:grid.Ny; (blockIdx().y - 1) * blockDim().y + threadIdx().y)
-            @loop for i in (1:grid.Nx; (blockIdx().x - 1) * blockDim().x + threadIdx().x)
-                @inbounds G⁻.Gu[i, j, k] = Gⁿ.Gu[i, j, k]
-                @inbounds G⁻.Gv[i, j, k] = Gⁿ.Gv[i, j, k]
-                @inbounds G⁻.Gw[i, j, k] = Gⁿ.Gw[i, j, k]
-                @inbounds G⁻.GT[i, j, k] = Gⁿ.GT[i, j, k]
-                @inbounds G⁻.GS[i, j, k] = Gⁿ.GS[i, j, k]
-            end
-        end
-    end
-end
-
-"""
-Update the hydrostatic pressure perturbation pHY′. This is done by integrating
-the `buoyancy_perturbation` downwards:
-
-    `pHY′ = ∫ buoyancy_perturbation dz` from `z=0` down to `z=-Lz`
-"""
-function update_hydrostatic_pressure!(pHY′, grid, buoyancy, Φ)
-    @loop for j in (1:grid.Ny; (blockIdx().y - 1) * blockDim().y + threadIdx().y)
-        @loop for i in (1:grid.Nx; (blockIdx().x - 1) * blockDim().x + threadIdx().x)
-            @inbounds pHY′[i, j, 1] = - ▶z_aaf(i, j, 1, grid, buoyancy_perturbation, buoyancy, Φ) * grid.Δz
-            @unroll for k in 2:grid.Nz
-                @inbounds pHY′[i, j, k] = 
-                    pHY′[i, j, k-1] - ▶z_aaf(i, j, k, grid, buoyancy_perturbation, buoyancy, Φ) * grid.Δz
-            end
-        end
-    end
-end
+#####
+##### Navier-Stokes and tracer advection equations
+#####
 
 """ Calculate the right-hand-side of the u-momentum equation. """
-function calculate_Gu!(Gu, grid, coriolis, closure, U, Φ, pHY′, K, F, parameters, time)
+function calculate_Gu!(Gu, grid, coriolis, closure, U, C, K, F, pHY′, parameters, time)
```

Comment:
Should `C` be our notation for all tracers? I feel like we were using `Φ` but I'm happy with both. Actually `C` might be better as `ϕ` is already used for kinematic pressure.

Reply:
The conflict with our notation for kinematic pressure is the reason I made this change.